### PR TITLE
fix: spending-limit save 422 and accidental dialog dismiss

### PR DESCRIPTION
**Pull request by d-mo**
## Summary
- Next steps "Set a spending limit" posted `notification_user_ids: [null]` because `/api/v1/auth/users/me` has no `id`. The editor now matches the current user from `/users` by email, or omits recipients. Empty/null ids are stripped before POST.
- Overlay clicks no longer close the spending-limits dialog. Hoisted notify/period dropdowns portal outside the panel, so a click on an option was treated as an overlay dismiss.

## Test plan
- [ ] Overview Next steps → Set a spending limit → Add limit → save a monthly hard cap (notify off). Expect 201, not 422.
- [ ] Enable Notify on soft, open the recipients dropdown, click a user. Dialog stays open. Save succeeds.
- [ ] Escape and the X still close the dialog.
- [ ] `budget-policy-editor`, `budget-limits-dialog`, `notify-recipients-field` tests (18 passed locally).

> [!NOTE]
> **[Low Risk]**
> Well-tested frontend-only bug fix with no security implications; only a minor edge-case observation.
>
> **Overview**
> Fixes a spending-limit save 422 (was posting `notification_user_ids: [null]`) by resolving the current user from `/users` by email and stripping empty ids, and stops overlay clicks from dismissing the limits dialog so hoisted dropdowns don't close it.
>
> <sup>Written by [Preloop PR Reviewer](https://preloop.ai) for commit 752c8d43. Updates automatically on new commits.</sup>

### CODE REVIEW

File: frontend/src/components/budget-policy-editor.ts [hunk 1]
```diff
@@ -452,23 +455,46 @@ export class BudgetPolicyEditor extends LitElement {
...
+    userProfile: { id?: unknown; email?: string } | null
+  ): string | undefined {
+    if (typeof userProfile?.id === 'string' && userProfile.id.length > 0) {
+      return userProfile.id;
+    }
+    const email = userProfile?.email;
+    if (!email) {
+      return undefined;
+    }
+    return this.availableUsers.find((user) => user.email === email)?.id;
```
[the remark this answers is not in the document]
Comment (d-mo) [reply frontend/src/components/budget-policy-editor.ts]: Matched case-insensitively against the users list. `/auth/users/me` still has no `id`; adding it is a separate schema/OpenAPI change. Accounts over 100 users can still miss the default recipient, same as before, rather than posting `[null]`.
